### PR TITLE
death to format

### PR DESCRIPTION
**Pull request by rdallman**
that-s-right-death-to-smoochy-39352604-608-352

this is the end of the line for format. what was previously 'http-stream' format is now the only format. this completes the prophecy. the cli can get updated after this to drop the field, as well. Hopefully a bulleted list will clarify what happened here in order to verify it, if not, sorry, I tried. To the details:

__NOTE:__ https://github.com/fnproject/fdk-go/pull/39 is a prerequisite for this PR (if you don't believe in committing branch revision shas to the Gopkg.lock file here, like me), please review that if it remains unmerged at time of reading this, if there's any changes there they'll need to be picked up here.

* removed `fn.format` field from API (and swagger)
* added migration and removed `fn.format` field from db
* removed all `protocol` packages, formats and associated references
* removed cold slots from agent (and default format)
* fix up fn-test-utils image for our tests, to fdk w/ only http-stream
* updates fdk-go dep for fn-test-utils image (reminder: we need to update to master w/ the PR linked above for fdk-go merged)
* removed all tests for other formats in server tests, changed agent tests to http-stream only
* removed 'hello' image we were maintaining here - it was 'default', the fdk-go has an example too
* removed references to 'cold' things. it's feeling tropical in here now.
* updated system-tests to fail when they fail (hint: must be in main goroutine), and to shut up unless they fail (really noisy in circleci)
* remove json/other content type detection on invoke/httptrigger endpoint output (yay)
* removed cloudevent detection on invoke/trigger - this is on fdks now
* remove headers: `Fn_call_id` and `Fn_deadline`, we have `Fn-Call-Id` and `Fn-Deadline` now
* remove FN_FORMAT, FN_APP_NAME, FN_PATH env vars (path was from route, we could add FN_TRIGGER_ID to headers maybe? they have req url tho). add FN_APP_ID env var.
* turns off the stdout/stderr blocking between calls (if freezer is on), because it's
impossible to debug without that (without syslog), now that stdout and stderr
go to the same place (either to host stderr or nowhere) and isn't used for
function output this shouldn't be a big fuss really. this helps debugging. a lot.
and in prod these are going to /dev/null anyway.
* fix agent call creation to insert FN_LISTENER for any calls created (fix status thing)
* fix NPE when giving a non-http.ResponseWriter to call as the writer (oops) - see `writeResp`
* remove stdin hookup in agent to docker, we don't need this anymore
* fixes bug where the watcher would leak if a container exited before entering the slot dealer goroutine (this does not change behavior like #1279 there's no overlap with that except in the diff, this just fixes a leak but the behavior remains unchanged in the 'happy' path, no new timeouts, etc) - moved the `shutdownContainer` context in `runHot` up a little to enclose `inotifyWait`

__questions/requests for you, o benevolent reviewer:__

* should we remove fn-test-utils and fn-status-checker Gopkg.lock files? they’re redundant. or we should move to a different repo maybe. will have to update here after fdk-go changes land, anyway.
* we can probably remove a lot of the funky pipe business, but I chose not to do that here. it's some amount of gutting we can do later, it doesn't effect this really (maybe some perf bump, idk).
* please check that I didn’t miss any tests when migrating, there were lots of dupes for each format and now there’s just 1 of each. I double checked, but I miss things sometimes.
* no @mantree I can't split it up into multiple PRs :)

### CODE REVIEW

**Comment by zootalures** [comment]
Code Looks good and full of wholesome deletage  - have tried and tested it a bit and I see no errors

**Comment by zootalures** [comment]
It looks like old non-stream functions fail with a 500 on invoke - should  this to be something else or something more informative? 

```
time="2018-10-25T16:41:09Z" level=error msg="internal server error" action="server.handleFnInvokeCall)-fm" error="context canceled" fnID=01CTP1KQ29NG8G00GZJ0000002 stack="goroutine 100 [running]:\nruntime/debug.Stack(0xc4200ad900, 0x1213a20, 0xc42009c1c0)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0xa7\ngithub.com/fnproject/fn/api/server.HandleErrorResponse(0x121f960, 0xc420441950, 0x7f042ee997d8, 0xc4200e82c0, 0x1213a20, 0xc42009c1c0)\n\t/go/src/github.com/fnproject/fn/api/server/error_response.go:52 +0x4b2\ngithub.com/fnproject/fn/api/server.handleErrorResponse(0xc4200e82c0, 0x1213a20, 0xc42009c1c0)\n\t/go/src/github.com/fnproject/fn/api/server/error_response.go:24 +0xa9\ngithub.com/fnproject/fn/api/server.(*Server).handleFnInvokeCall(0xc42027f8c0, 0xc4200e82c0)\n\t/go/src/github.com/fnproject/fn/api/server/runner_fninvoke.go:45 +0x1da\ngithub.com/fnproject/fn/api/server.(*Server).(github.com/fnproject/fn/api/server.handleFnInvokeCall)-fm(0xc4200e82c0)\n\t/go/src/github.com/fnproject/fn/api/server/server.go:1169 +0x34\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Context).Next(0xc4200e82c0)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/context.go:108 +0x43\ngithub.com/fnproject/fn/api/server.(*Server).runMiddleware(0xc42027f8c0, 0xc4200e82c0, 0x0, 0x0, 0x0)\n\t/go/src/github.com/fnproject/fn/api/server/middleware.go:42 +0x29f\ngithub.com/fnproject/fn/api/server.(*Server).rootMiddlewareWrapper.func1(0xc4200e82c0)\n\t/go/src/github.com/fnproject/fn/api/server/middleware.go:32 +0x58\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Context).Next(0xc4200e82c0)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/context.go:108 +0x43\ngithub.com/fnproject/fn/api/server.apiMetricsWrap.func1.1(0xc4200e82c0)\n\t/go/src/github.com/fnproject/fn/api/server/gin_middlewares.go:153 +0x2f7\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Context).Next(0xc4200e82c0)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/context.go:108 +0x43\ngithub.com/fnproject/fn/api/server.panicWrap(0xc4200e82c0)\n\t/go/src/github.com/fnproject/fn/api/server/gin_middlewares.go:186 +0x51\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Context).Next(0xc4200e82c0)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/context.go:108 +0x43\ngithub.com/fnproject/fn/api/server.traceWrap(0xc4200e82c0)\n\t/go/src/github.com/fnproject/fn/api/server/gin_middlewares.go:96 +0x507\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Context).Next(0xc4200e82c0)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/context.go:108 +0x43\ngithub.com/fnproject/fn/api/server.loggerWrap(0xc4200e82c0)\n\t/go/src/github.com/fnproject/fn/api/server/gin_middlewares.go:208 +0x1ea\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Context).Next(0xc4200e82c0)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/context.go:108 +0x43\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Engine).handleHTTPRequest(0xc42027f7a0, 0xc4200e82c0)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/gin.go:361 +0x586\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Engine).ServeHTTP(0xc42027f7a0, 0x121f2e0, 0xc4200ad860, 0xc420209f00)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/gin.go:326 +0x153\ngithub.com/fnproject/fn/vendor/go.opencensus.io/plugin/ochttp.(*Handler).ServeHTTP(0xc4201d3c40, 0x121f2e0, 0xc4200ad860, 0xc420153700)\n\t/go/src/github.com/fnproject/fn/vendor/go.opencensus.io/plugin/ochttp/server.go:82 +0x1c8\nnet/http.serverHandler.ServeHTTP(0xc420405e10, 0x121dfa0, 0xc420478380, 0xc420153700)\n\t/usr/local/go/src/net/http/server.go:2694 +0xbc\nnet/http.(*conn).serve(0xc420224000, 0x121f8a0, 0xc4200ae3c0)\n\t/usr/local/go/src/net/http/server.go:1830 +0x651\ncreated by net/http.(*Server).Serve\n\t/usr/local/go/src/net/http/server.go:2795 +0x27b\n"
time="2018-10-25T16:52:00Z" level=info msg="hot function terminated" app_id=01CT5ZVBCZNG8G00GZJ000000N cpus= error="container exit code 126" fn_id=01CTP26YZPNG8G00GZJ0000006 id=01CTP27KBWNG8G00GZJ0000009 idle_timeout=30 image="foo:0.0.2" memory=128
time="2018-10-25T16:52:00Z" level=error msg="internal server error" action="server.handleFnInvokeCall)-fm" error="context canceled" fnID=01CTP26YZPNG8G00GZJ0000006 stack="goroutine 2131 [running]:\nruntime/debug.Stack(0xc420334370, 0x1213a20, 0xc42009c1c0)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0xa7\ngithub.com/fnproject/fn/api/server.HandleErrorResponse(0x121f960, 0xc4204fadb0, 0x7f042ee997d8, 0xc4200e8370, 0x1213a20, 0xc42009c1c0)\n\t/go/src/github.com/fnproject/fn/api/server/error_response.go:52 +0x4b2\ngithub.com/fnproject/fn/api/server.handleErrorResponse(0xc4200e8370, 0x1213a20, 0xc42009c1c0)\n\t/go/src/github.com/fnproject/fn/api/server/error_response.go:24 +0xa9\ngithub.com/fnproject/fn/api/server.(*Server).handleFnInvokeCall(0xc42027f8c0, 0xc4200e8370)\n\t/go/src/github.com/fnproject/fn/api/server/runner_fninvoke.go:45 +0x1da\ngithub.com/fnproject/fn/api/server.(*Server).(github.com/fnproject/fn/api/server.handleFnInvokeCall)-fm(0xc4200e8370)\n\t/go/src/github.com/fnproject/fn/api/server/server.go:1169 +0x34\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Context).Next(0xc4200e8370)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/context.go:108 +0x43\ngithub.com/fnproject/fn/api/server.(*Server).runMiddleware(0xc42027f8c0, 0xc4200e8370, 0x0, 0x0, 0x0)\n\t/go/src/github.com/fnproject/fn/api/server/middleware.go:42 +0x29f\ngithub.com/fnproject/fn/api/server.(*Server).rootMiddlewareWrapper.func1(0xc4200e8370)\n\t/go/src/github.com/fnproject/fn/api/server/middleware.go:32 +0x58\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Context).Next(0xc4200e8370)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/context.go:108 +0x43\ngithub.com/fnproject/fn/api/server.apiMetricsWrap.func1.1(0xc4200e8370)\n\t/go/src/github.com/fnproject/fn/api/server/gin_middlewares.go:153 +0x2f7\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Context).Next(0xc4200e8370)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/context.go:108 +0x43\ngithub.com/fnproject/fn/api/server.panicWrap(0xc4200e8370)\n\t/go/src/github.com/fnproject/fn/api/server/gin_middlewares.go:186 +0x51\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Context).Next(0xc4200e8370)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/context.go:108 +0x43\ngithub.com/fnproject/fn/api/server.traceWrap(0xc4200e8370)\n\t/go/src/github.com/fnproject/fn/api/server/gin_middlewares.go:96 +0x507\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Context).Next(0xc4200e8370)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/context.go:108 +0x43\ngithub.com/fnproject/fn/api/server.loggerWrap(0xc4200e8370)\n\t/go/src/github.com/fnproject/fn/api/server/gin_middlewares.go:208 +0x1ea\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Context).Next(0xc4200e8370)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/context.go:108 +0x43\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Engine).handleHTTPRequest(0xc42027f7a0, 0xc4200e8370)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/gin.go:361 +0x586\ngithub.com/fnproject/fn/vendor/github.com/gin-gonic/gin.(*Engine).ServeHTTP(0xc42027f7a0, 0x121f2e0, 0xc4203342d0, 0xc420153600)\n\t/go/src/github.com/fnproject/fn/vendor/github.com/gin-gonic/gin/gin.go:326 +0x153\ngithub.com/fnproject/fn/vendor/go.opencensus.io/plugin/ochttp.(*Handler).ServeHTTP(0xc4201d3c40, 0x121f2e0, 0xc4203342d0, 0xc420153400)\n\t/go/src/github.com/fnproject/fn/vendor/go.opencensus.io/plugin/ochttp/server.go:82 +0x1c8\nnet/http.serverHandler.ServeHTTP(0xc420405e10, 0x121dfa0, 0xc42043b420, 0xc420153400)\n\t/usr/local/go/src/net/http/server.go:2694 +0xbc\nnet/http.(*conn).serve(0xc420224000, 0x121f8a0, 0xc4201d3240)\n\t/usr/local/go/src/net/http/server.go:1830 +0x651\ncreated by net/http.(*Server).Serve\n\t/usr/local/go/src/net/http/server.go:2795 +0x27b\n"

```

```
POST /invoke/01CTP26YZPNG8G00GZJ0000006 HTTP/1.1
Host: localhost:8080
User-Agent: Go-http-client/1.1
Content-Length: 0
Content-Type: text/plain
Accept-Encoding: gzip


HTTP/1.1 500 Internal Server Error
Content-Length: 36
Content-Type: application/json; charset=utf-8
Date: Thu, 25 Oct 2018 16:52:00 GMT

{"message":"internal server error"}

Error:

Fn: Error calling function: status 500
```

File: Makefile [hunk 1]
```diff
@@ -74,10 +74,6 @@ test-system: test-basic
 img-busybox:
 	docker pull busybox
 
-.PHONY: img-hello
-img-hello:
-	docker pull fnproject/hello
-
```
Comment (skinowski) [inline Makefile]: :1st_place_medal:

**Comment by rdallman** [comment]
> deletage

the internet doesn't think this is a word. I'm curious now :)

**Comment by zootalures** [comment]
It breaks all java based FDKs so, err no

**Comment by zootalures** [comment]
Unblocking but can you re-add FN_FORMAT for a bit to make java happy

**Comment by denismakogon** [comment]
Ok, that may work.

**Comment by rdallman** [comment]
yea, adding back FN_FORMAT and trying to fix up the error now.

**Comment by zootalures** [comment]
Here is the ClI set I was testing against: 
https://github.com/fnproject/cli/pull/472

**Comment by zootalures** [comment]
| | deletage
| the internet doesn't think this is a word. I'm curious now :)

It's medieval french - the state of some software having benefited from the removal of some of its code. 

"Mon dieu, regarde le deletage de ce pull-request"

File: api/agent/agent.go [hunk 2]
```diff
@@ -1045,7 +822,10 @@ func (a *agent) runHot(ctx context.Context, call *call, tok ResourceToken, state
 			}
 
 		case <-ctx.Done():
-			call.slots.queueSlot(&hotSlot{done: make(chan struct{}), fatalErr: ctx.Err()})
+			// XXX(reed): this seems like a bad idea? why are we even handing out a
+			// bad slot? shouldn't we make the client wait for a valid one and maybe
+			// timeout? not in this PR, NOT TONIGHT!
+			call.slots.queueSlot(&hotSlot{done: make(chan struct{}), fatalErr: models.ErrContainerExitedEarly})
```
Comment (rdallman) [inline api/agent/agent.go]: @skinowski we should discuss this behavior maybe somewhere else, this PR is a stinky enough turd without getting into the weeds on this behavior here, just highlighting...

**Comment by rdallman** [comment]
see https://github.com/fnproject/fn/pull/1281/commits/c425c1db2cbde82ad73d5763de6d7c3c897643ba and https://github.com/fnproject/fn/pull/1281/commits/0a703e7a66a01953d9b6ba5212b4371debe928dd to add an informative error for the explosive fdks and FN_FORMAT env var to http-stream during transition for fdks. I did not add a test for the latter, I could have another go at updating the test utils image but I'm about 99% positive it works and shouldn't be critical anyway and I smell food which is much more critical...

**Comment by rdallman** [comment]
sickcloseazurevase-size_restricted

**Comment by rdallman** [comment]
thanks all

**Pull request merged**